pkg/api/upload.go

```diff
@@ -110,14 +113,46 @@ func (x UploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if r.Header.Get("Content-Type") != "application/json" {
-		invalidContent(w)
-		return
+	var body bytes.Buffer
+	switch r.Header.Get("Content-Encoding") {
+	case "application/gzip":
+		gzipReader, err := gzip.NewReader(r.Body)
+		if err != nil {
+			logger.WithError(err).Error("gzip.NewReader() failed")
+			badRequest(w, "")
+		}
+		if _, err := body.ReadFrom(gzipReader); err != nil {
+			logger.WithError(err).Error("gzipReader.Read() failed")
+			badRequest(w, "")
+		}
+		if err := gzipReader.Close(); err != nil {
+			logger.WithError(err).Error("gzipReader.Close() failed")
+			badRequest(w, "")
+		}
+
+	default:
+		if _, err := body.ReadFrom(r.Body); err != nil {
+			logger.WithError(err).Error("gzip.Read() failed")
+			badRequest(w, "")
+		}
 	}
```

Comment:
Is it really necessary to fail on no gzip?

Reply:
I oh I think I need to just update the log message here. This isn't actually failing if not gzip.